addons/hr_expense/tests/test_expenses.py

```diff
@@ -1179,3 +1181,86 @@ def test_expense_from_attachments(self):
+                'res_id': expense_with_attachment_2.id,
+                'res_model': 'hr.expense',
+            }, {
+                'name': "test_file_3.png",
+                'datas': base64.b64encode(b'different content'),
+                'res_id': expense_with_attachment_3.id,
+                'res_model': 'hr.expense',
+            }])
+
+            expense_sheet = self.env['hr.expense.sheet'].create({
+                'company_id': self.env.company.id,
+                'employee_id': self.expense_employee.id,
+                'name': 'test sheet',
+                'expense_line_ids': [Command.set([expense_with_attachment_1.id, expense_with_attachment_2.id, expense_with_attachment_3.id])],
+            })
+
+            sheet_attachment = self.env['ir.attachment'].create({
+                'name': "test_file_4.png",
+                'datas': base64.b64encode(b'yet another different content'),
+                'res_id': expense_sheet.id,
+                'res_model': 'hr.expense.sheet',
+            }) if sheet_has_attachment else self.env['ir.attachment']
+
+            assert_attachments_are_synced(expense_sheet.expense_line_ids.attachment_ids, expense_sheet.attachment_ids, sheet_attachment, sheet_has_attachment)
+            expense_with_attachment_1.attachment_ids.unlink()
+            assert_attachments_are_synced(expense_sheet.expense_line_ids.attachment_ids, expense_sheet.attachment_ids, sheet_attachment, sheet_has_attachment)
+            self.env['ir.attachment'].create({
+                'name': "test_file_1.png",
+                'datas': base64.b64encode(b'content'),
+                'res_id': expense_with_attachment_1.id,
+                'res_model': 'hr.expense',
+            })
+            assert_attachments_are_synced(expense_sheet.expense_line_ids.attachment_ids, expense_sheet.attachment_ids, sheet_attachment, sheet_has_attachment)
+            expense_with_attachment_2.sheet_id = False
+            assert_attachments_are_synced(expense_sheet.expense_line_ids.attachment_ids, expense_sheet.attachment_ids, sheet_attachment, sheet_has_attachment)
+            expense_with_attachment_2.sheet_id = expense_sheet
+            assert_attachments_are_synced(expense_sheet.expense_line_ids.attachment_ids, expense_sheet.attachment_ids, sheet_attachment, sheet_has_attachment)
+            expense_sheet.write({'expense_line_ids': [Command.set([expense_with_attachment_1.id, expense_with_attachment_3.id])]})
+            assert_attachments_are_synced(expense_sheet.expense_line_ids.attachment_ids, expense_sheet.attachment_ids, sheet_attachment, sheet_has_attachment)
+            expense_with_attachment_3.unlink()
+            assert_attachments_are_synced(expense_sheet.expense_line_ids.attachment_ids, expense_sheet.attachment_ids, sheet_attachment, sheet_has_attachment)
+            expense_sheet.attachment_ids.filtered(lambda att: att.checksum in expense_sheet.expense_line_ids.attachment_ids.mapped('checksum')).unlink()
```

Comment:
indent like this for instance:
```suggestion
            sheet.attachment_ids.filtered(
                lambda att: att.checksum in sheet.expense_line_ids.attachment_ids.mapped('checksum')
            ).unlink()
```